tests/functional/demo/test_demo.py

```diff
@@ -243,30 +241,84 @@ def test_demo_inexecutable_script(storage, monkeypatch, capsys):
     assert "User script is not executable" in captured
 
 
-def test_demo_four_workers(storage, monkeypatch):
+@contextmanager
+def generate_config(template):
+    """Generate a configuration file inside a temporary directory with the current storage config"""
+    tmp_path = tempfile.mkdtemp("orion_test")
```

Comment:
If it relied on `tmp_path` as argument, passed from Pytest's `tmp_path`, that would ensure the folder is cleared even when the test fails.

Reply:
done